sea-orm-migration/src/migrator.rs

```diff
@@ -99,36 +106,56 @@ pub trait MigratorTrait: Send {
     }
 
     /// Get list of pending migrations
-    async fn get_pending_migrations(db: &DbConn) -> Result<Vec<Migration>, DbErr> {
-        Self::install(db).await?;
-        Ok(Self::get_migration_with_status(db)
+    async fn get_pending_migrations<T: MigrationConnectionTrait>(
+        conn: &T,
+    ) -> Result<Vec<Migration>, DbErr> {
+        Self::install(conn).await?;
+        Ok(Self::get_migration_with_status(conn)
             .await?
             .into_iter()
             .filter(|file| file.status == MigrationStatus::Pending)
             .collect())
     }
 
     /// Get list of applied migrations
-    async fn get_applied_migrations(db: &DbConn) -> Result<Vec<Migration>, DbErr> {
-        Self::install(db).await?;
-        Ok(Self::get_migration_with_status(db)
+    async fn get_applied_migrations<T: MigrationConnectionTrait>(
+        conn: &T,
+    ) -> Result<Vec<Migration>, DbErr> {
+        Self::install(conn).await?;
+        Ok(Self::get_migration_with_status(conn)
             .await?
             .into_iter()
             .filter(|file| file.status == MigrationStatus::Applied)
             .collect())
     }
 
     /// Create migration table `seaql_migrations` in the database
-    async fn install(db: &DbConn) -> Result<(), DbErr> {
+    async fn install<T: MigrationConnectionTrait>(conn: &T) -> Result<(), DbErr> {
+        let db = conn.get_connection();
         let builder = db.get_database_backend();
         let schema = Schema::new(builder);
         let mut stmt = schema.create_table_from_entity(seaql_migrations::Entity);
+
+        match conn.get_schema_name() {
+            Some(schema) => {
+                stmt.table = Some(
+                    (
+                        Alias::new(schema),
+                        Alias::new(seaql_migrations::Entity.table_name()),
+                    )
+                        .into_table_ref(),
+                )
+            }
+            None => {}
+        }
```

Comment:
@billy1624 Also I don't think this works since `TableCreateStatement.table` is only visible inside `sea-query` crate. So that means we'd need to modify `Schema.create_table_from_entity` method (which is also outside `sea-orm-migration`). Is this safe to modify (like by adding the `MigrationConnection` struct)?